hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -689,7 +689,7 @@ protected void compactIfNecessary(AbstractHoodieWriteClient writeClient, String
     String latestDeltacommitTime = metadataMetaClient.reloadActiveTimeline().getDeltaCommitTimeline().filterCompletedInstants().lastInstant()
         .get().getTimestamp();
     List<HoodieInstant> pendingInstants = dataMetaClient.reloadActiveTimeline().filterInflightsAndRequested()
-        .findInstantsBefore(latestDeltacommitTime).getInstants().collect(Collectors.toList());
+        .findInstantsBefore(instantTime).getInstants().collect(Collectors.toList());
```

Comment:
`compactionInstantTime` at line 703 has to be based off `instantTime` and not `latestDeltaCommitTime`. Latest delta commit time is not part of the compaction yet. Otherwise we are changing the meaning of the current compaction timeline with this change.

Reply:
let me try to explain. 
lets say we have 10 commits, C1, C2 -> C10. 
Prior to this patch, we will compact immediately after C10 and so compaction commit will be C10 + "001".

With this patch, we will be compacting just before C11 starts getting applied to MDT. 
And so, I am basing the compaction commit of latest delta commit time which is C10 and not instant time which is C11. 
And so, its C10 + "001". but if I go with instantTime, then we might change the behavior. In fact, we can't do that, since compaction time will be less than delta commit which will be eventually created when we apply C11 to MDT. 

Let me know if this makes sense.

Reply:
Right, I was actually asking the compaction time to be C10 and not C11. I misread line 689. Look good then.